firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -220,7 +225,12 @@ private final void doRegistration() {
 
       // New FID needs to be created
       if (persistedInstallationEntry.isNotGenerated()) {
-        String fid = utils.createRandomFid();
+
+        // Read the existing iid for this firebase installation
+        String fid = iidStore.readIid();
+        if (fid == null) {
+          fid = utils.createRandomFid();
+        }
```

Comment:
(please see my comments at the bottom)
Assuming you switch the IidStore class to avoid NULL values, you can rewrite this as
String fid = iidStroe.readIid().orElse(utils::createRandomFid);